src/merge_outputs.rs

```diff
@@ -124,9 +124,9 @@ pub async fn merge(
     }
 
     // External configs get merged first, then rules-based configs
-    if external_execution_config.is_some() {
+    for config in external_execution_configs {
         let mut external_base_cfg = ConfigDelta::default();
-        external_base_cfg.config = external_execution_config.clone();
+        external_base_cfg.config = Some(config.clone());
```

Comment:
I think we can remove this `.clone` if we make `external_execution_configs` not a reference.

Reply:
ah this was a nice catch